### DISCUSSION TITLE
DataGrid editing does not change Background Color and unreadable in Dark Mode

### DISCUSSION THREAD

**Discussion by NorMoll** [AvaloniaUI/Avalonia#16294] [Q&A]
### Describe the bug

When attepting to edit the DataGrid cell in Dark mode the text is unreadable as the Cell has a white background. Adding the following Style has no effect:

	<DataGrid.Styles>
		<Style Selector="DataGridCell:focus-within">
			<Setter Property="Background" Value="Red"/>
			<Setter Property="Foreground" Value="Black"/>
		</Style>
	</DataGrid.Styles>

This seemed to work on the first Column in 11.0.11 but as of 11.1.0-rc2 this doesnt wotk at all.

Thanks to the team for some great work though.

### To Reproduce
```
<DataGrid Name="grid"  ItemsSource="{Binding Cities}"				
		AutoGenerateColumns="False" IsReadOnly="False">
	<DataGrid.Styles>
		<Style Selector="DataGridCell:focus-within">
			<Setter Property="Background" Value="Red"/>
			<Setter Property="Foreground" Value="Black"/>
		</Style>
	</DataGrid.Styles>
	<DataGrid.Columns>
		<DataGridTextColumn Header="GeoNameID"  Binding="{Binding geonameid}" />
		<DataGridTextColumn Header="Name"  Binding="{Binding name}"/>
		<DataGridTextColumn Header="Country"  Binding="{Binding country}"/>
		<DataGridTextColumn Header="Subcountry"  Binding="{Binding subcountry}"/>
	</DataGrid.Columns>
</DataGrid>
```

### Avalonia version

11.0.11~11.1.0-rc2

### OS

Windows

**Comment by timunie** [comment]
Which theme so you use? Do you have some custom styles applied?

**Reply by NorMoll** [reply]
No Custom Themes applied.
App.axaml:
```
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="TestApp.App"             
	RequestedThemeVariant="Dark">
	<Application.Styles>		
		<FluentTheme  />
		<StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Fluent.xaml"/>
	</Application.Styles>
</Application>
```

To allow me to continue I've temporarily wrapped the grid in `<ThemeVariantScope RequestedThemeVariant="Light">`

**Reply by timunie** [reply]
If you remove your own styles for DataGridCell is it better? If so, the selector may be wrong. Lookup original styles of DataGrid to understand the selectors used

**Reply by NorMoll** [reply]
Removing the styles makes no difference. 

I created a new Avalonia 11 project and when editing the DataGrid I get the the expected behaviour when editing a Cell and the background remains black with the Text remaining white. However, in my original project with the identical App.axaml as the test app, upon selecting the row, as per the test app, turns blue but when editing the cell, the background immediately turns white rendering the white text unreadable. If I then select the text in the cell the text remains white but is highlighted in blue:

Also note that the ..focus-within as shown above has no effect in the test app, as when editing I expect the background to turn red and the text white but the background simply remains black.

![DataGridTest](https://github.com/user-attachments/assets/b8ccd09e-9bad-4b81-86e5-8d09838776e3)

**Reply by timunie** [reply]
That still means something in your App is fishy. Somewhere must be own Style or ControlTheme, be it in the MainView or MainWindow...

**Reply by NorMoll** [reply]
It was indeed a Style configured in the Page and the culprit was setting the TexBox:pointerover BackGround to White. Not sure why that was affecting a DataGrid Edit though.

```
<Style Selector="TextBox:pointerover /template/ Border#PART_BorderElement">
			<Setter Property="BorderBrush" Value="Black" />
			<Setter Property="Background" Value="White" />
</Style
```

**Reply by timunie** [reply]
The editing element is basically a TextBox. With some own styling. That's why your style could leak into. Use ControlTheme if you don't want that behavior or add a style to DataGrid which does reset the outer Style
